Parsers/ASimNetworkSession/ProductParsers/vimNetworkSessionMicrosoftWindowsFirewall.yaml

```diff
@@ -13,31 +13,164 @@ References:
+                                    '\x0d\x0a  <Data Name="SourcePort">'SourcePort'</Data>'
+                                    '\x0d\x0a  <Data Name="DestAddress">'DestAddress'</Data>'
+                                    '\x0d\x0a  <Data Name="DestPort">'DestPort'</Data>'
+                                    '\x0d\x0a  <Data Name="Protocol">'Protocol'</Data>'
+                                    '\x0d\x0a  <Data Name="FilterRTID">'NetworkRuleNumber'</Data>'
+                                    '\x0d\x0a  <Data Name="LayerName">'LayerCode'</Data>'
+                                    '\x0d\x0a  <Data Name="LayerRTID">'LayerRTID'</Data>'
+                                    '\x0d\x0a  <Data Name="RemoteUserID">'RemoteUserID'</Data>'
+                                    '\x0d\x0a  <Data Name="RemoteMachineID">'RemoteMachineID'</Data>'*;
+            union isfuzzy=true SecurityEvent_5152, SecurityEvent_5156_5157, SecurityEvent_5154_5155_5158_5159 | extend toint(Protocol)
+            | lookup Directions on DirectionCode
+            | extend // need to fix this extend -> need to take the column name out
+                  SrcAppName = iff(isOutBound, Application, ""),
+                  DstAppName = iff(not(isOutBound), Application, ""),
+                  SrcIpAddr = iff(isOutBound, DestAddress, SourceAddress),
+                  DstIpAddr = iff(not(isOutBound), SourceAddress, DestAddress),
+                  SrcHostId = iff(isOutBound, RemoteMachineID, ""),
+                  DstHostId = iff(not(isOutBound), RemoteMachineID, ""),
+                  SrcPortNumber = iff(isOutBound, toint(DestPort), toint(SourcePort)),
+                  DstPortNumber = iff(not(isOutBound), toint(SourcePort), toint(DestPort)),
+                  SrcProcessId =  iff(isOutBound, toint(ProcessId), toint("")),
+                  DstProcessId =  iff(not(isOutBound), toint(ProcessId), toint("")),
+                  DstUserId = iff(isOutBound, RemoteUserID, ""),
+                  SrcUserId = iff(not(isOutBound), RemoteUserID, "")
+            };
+            //////////////////////////////////////////////////////
+            // this query extract the data from WindowsEvent table
+            //////////////////////////////////////////////////////
+            let WindowsFirewall_WindowsEvent=(){   
+                      WindowsEvent
+                      | where EventID between (5150 .. 5159)
+                      | extend 
+                          EventSeverity=tostring(EventData.Severity),
+                          LayerCode = tostring(EventData.LayerName),
+                          NetworkRuleNumber = tostring(EventData.FilterRTID),
+                          Protocol = toint(EventData.Protocol),
+                          DirectionCode = iff(EventID in (5154, 5155, 5158, 5159), "%%14609",tostring(EventData.Direction))
+                      | extend iff(EventID in (5154, 5155, 5158, 5159), DirectionCode="%%14609",DirectionCode = tostring(EventData.Direction) )
```

Comment:
I believe this line is a leftover.

Reply:
no, because "Listen" events don't have direction field this is how im creating it

Reply:
Not line 137 but rather line 138.... it does nothing.